src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -308,11 +309,34 @@ FindDataObjectsRequest buildRequestHash(int limit) {
          * @return object encapsulating the result set
          */
         public FindDataObjectsResult<T> execute(int pageSize) {
-            return new FindDataObjectsResult<T>(this.buildRequestHash(pageSize), this.classConstraint,
+            return new FindDataObjectsResult<T>(this.buildRequestHash(pageSize, null), this.classConstraint,
                     this.env, pageSize);
         }
 
         /**
+         * Executes the query and returns first result set page.
+         *
+         * @param pageSize Page size
+         *
+         * @return Iterable result set page
+         */
+        public SearchPage<T> getFirstPage(int pageSize) {
+            return new SearchPage<T>(this.buildRequestHash(pageSize, null), this.classConstraint, this.env);
+        }
+
+        /**
+         * Executes the query and returns subsequent result set page.
+         *
+         * @param pageSize Page size
+         * @param starting Result of SearchPage<T>.getNext() call on previous page
+         *
+         * @return Iterable result set page
+         */
+        public SearchPage<T> getSubsequentPage(int pageSize, JsonNode starting) {
+            return new SearchPage<T>(this.buildRequestHash(pageSize, starting), this.classConstraint, this.env);
```

Comment:
In this code path, assert that starting is not null.

e.g. Preconditions.checkNotNull